examples/typescript/platform-ref-aws/crossplane.ts

```diff
@@ -0,0 +1,125 @@
+  });
+
+  config.addProvider('registry.upbound.io/crossplane/provider-aws', '>=v0.14.0-0');
+  config.addProvider('registry.upbound.io/crossplane/provider-helm', '>=v0.3.6-0');
+
+  return yaml;
+}
+
+export function postgresYaml(crossplanePackage: App): Chart {
+  const yaml = new Chart(crossplanePackage, 'postgres-api');
+  const xrd = new CompositeResourceDefinition(yaml, 'postgres-xrd', {
+    name: 'compositepostgresqlinstances.aws.platformref.crossplane.io',
+  });
+
+  xrd.group('aws.platformref.crossplane.io');
+  xrd.claimKind('PostgreSQLInstance').plural('postgresqlinstances');
+  xrd.kind('CompositePostgreSQLInstance').plural('compositepostgresqlinstances');
+  xrd.connectionSecret().key('username').key('password').key('endpoint').key('port');
+  xrd.version('v1alpha1').served().referencable()
```

Comment:
The fluent api is nice but we usually try to avoid it in CDKs because it doesn't feel natural in all supported languages. We prefer simple OO imperative APIs in general as they are also usually easier to understand for a broader audience

Reply:
Interesting ... as designed it can work either way, but perhaps not as clean as xrd.connectionSecret.addKey() for simple OO, but much easier to read with the fluent API. Let's chat about this tomorrow.